### ISSUE TITLE
ossec-syscheckd: ERROR: Unable to run diff

### ISSUE THREAD

**Issue by iodisciple** [ossec/ossec-hids#1296]
Hi,

I'm getting a lot of these in the /var/ossec/logs/ossec.log:

2017/10/26 09:43:06 ossec-syscheckd: ERROR: Unable to run `diff "/var/ossec/tmp/syscheck-changes-8f6defa4d18660368b1066d74c897b0e-1509003786" "/var/ossec/tmp/syscheck-changes-2ba80a71a76ed117877995814c0c3437-1508512101" > "/var/ossec/tmp/syscheck-changes-2ba80a71a76ed117877995814c0c3437-1508512101-8f6defa4d18660368b1066d74c897b0e-1509003786" 2> /dev/null`

OSSEC seems to be running fine though. I'm on CentOS 7.4. OSSEC is at 2.9.2. The errors were also on OSSEC 2.8.3. I temporarily disabled SELinux but the problem remained.

Any ideas?

Thanks and kind regards,
iodisciple

**Comment by iodisciple** [comment]
Any ideas what is causing this / tips how to solve this?

**Comment by ddpbsd** [comment]
Is there space on the partition OSSEC resides on? Enough inodes? Does `diff` exist on the system?
I think the error message needs some work.

**Comment by iodisciple** [comment]
Yes there's space, enough inodes (I've been there and this resulted in different errors) and diff exists.

**Comment by zvanderbilt** [comment]
Dont quote me but I am pretty sure you can get these errors when you delete files that had been watched by syscheck. it cant diff because the files are gone. I always get a ton of those errors after I delete a swath of files.

**Comment by iodisciple** [comment]
When I delete a file, I get an email alert but not this error in the log. I did think you were on to something since around the time I got two rsync errors about vanished files, I got two erros in the OSSEC log as well. But: the rsync directory in question is excluded in my OSSEC conf.

Thanks for your feedback, I'm thinking about just leaving it at this. The system seems to be working normally...

**Comment by VtG242** [comment]
I'm using ossec-3.0.0 and I also see time to time this error messages on some hosts.
Everything seems to be working as usually next syscheck scan ends OK but still I'm curious what's wrong. Any idea?

**Comment by krishnaprasanth** [comment]
Same for me. I have installed ossec-hids-3.3.0  and running it in "local" mode. I see these errors. Would be good to understand about this and the mitigation.
